submissions/management/commands/analyze_uploaded_file_sizes.py

```diff
@@ -0,0 +1,157 @@
+                          Defaults to <today - 30 days>.
+                          The maximum value for this argument is 2019-11-01.
+                          Before this day, file size data is unavailable.
+
+    both arguments must be in the form yyyy-mm-dd
+"""
+
+from __future__ import absolute_import
+
+import datetime
+import json
+
+from django.core.management.base import BaseCommand, CommandError
+
+from submissions.models import Submission
+
+DATE_FORMAT = '%Y-%m-%d'
+LAST_DATE = datetime.date(2019, 11, 1)
+HEADER = u'Course ID\tUsers with Uploaded Files\tTotal Uploaded Bytes\tAverage Upload per User'
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+
+        ./manage.py lms analyze_uploaded_file_sizes --start_date=2020-01-30 --end_date=2019-12-31
+
+    """
+    help = u'Collects and prints stats about ORA file upload usage'
+
+    def add_arguments(self, parser):
+        """
+        Add arguments to the command parser.
+        """
+        today = datetime.date.today()
+        thirty_days = datetime.timedelta(days=30)
+        thirty_days_ago = today - thirty_days
+
+        def parse_date(s):
+            datetime.datetime.strptime(s, DATE_FORMAT).date()
+
+        parser.add_argument(
+            '--start_date',
+            type=parse_date,
+            default=today,
+            help=u"The most recent day to include in the report. Defaults to today."
+        )
+        parser.add_argument(
+            '--end_date',
+            type=parse_date,
+            default=thirty_days_ago,
+            help=(u"The longest-ago day to include in the report. Defaults to <today - 30 days>. "
+                  u"Cannot be set to before 2019-11-01.")
+        )
+
+    # pylint: disable=arguments-differ, unused-argument
+    def handle(self, start_date, end_date, *args, **options):
+        """
+        Analyze ORA file upload submissions and and print tab-limited report
+        """
+        print(u'Starting file upload submission report')
+        arg_echo_str = u'start_date = {}, end_date = {}'.format(
+            start_date.strftime(DATE_FORMAT),
+            end_date.strftime(DATE_FORMAT)
+        )
+        print(arg_echo_str)
+        end_date = self.validate_input_dates(start_date, end_date)
+        submission_data = self.load_data(start_date, end_date)
+        self.print_report(submission_data)
+
+    def validate_input_dates(self, start_date, end_date):
+        """
+        Checks the validity of start_date and end_date
+        Checks that the end date is not before LAST_DATE, and if it is, overrides it.
+        """
+        if start_date < end_date:
+            raise CommandError("End date must be before start date")
+        if datetime.date.today() < start_date:
+            print(u"Warning: start_date is in the future")
+        if end_date < LAST_DATE:
+            print(u'File size data is unavailable before 2019-11-01. Setting end_date to 2019-11-01')
```

Comment:
Nit: use format string with the `LAST_DATE` constant (or whatever you might rename it to).